packages/medusa/src/services/product-variant.ts

```diff
@@ -348,13 +374,16 @@ class ProductVariantService extends BaseService {
         this.moneyAmountRepository_
       )
 
-      let moneyAmount = await moneyAmountRepo.findOne({
-        where: {
-          currency_code: price.currency_code?.toLowerCase(),
+      let moneyAmount = await moneyAmountRepo
+        .createQueryBuilder("money_amount")
+        .where("money_amount.currency_code = :currency_code", {
+          currency_code: price.currency_code,
+        })
+        .andWhere("money_amount.variant_id = :variant_id", {
           variant_id: variantId,
-          region_id: IsNull(),
-        },
-      })
+        })
+        .andWhere("money_amount.region_id IS NULL")
+        .getOne()
```

Comment:
Let's also move this into the repo. We can potentially make it even more powerful so it functions more like an upsert in the repository layer and then we can name the function something like: `upsertCurrencyPrice`? This way we don't have to do the query and decide pattern in the service layer which gets a bit tedious.